custom_connectors/basic_auth/stripe_connector.rb

```diff
@@ -0,0 +1,920 @@
+        get("/v1/customers?limit=1")["data"].first || {}
+      end
+    },
+
+    get_charge: {
+      description: "Get <span class='provider'>charges</span> details in " \
+        "<span class='provider'>Stripe</span>",
+
+      input_fields: lambda do
+        [
+          {
+            name: "id",
+            label: "Charge ID",
+            optional: false
+          }
+        ]
+      end,
+
+      execute: lambda do |_connection, input|
+        get("/v1/charges/#{input['id']}")
+      end,
+
+      output_fields: lambda do |object_definitions|
+        object_definitions["charge"]
+      end,
+
+      sample_output: lambda do |_connection|
+        get("/v1/charges?limit=1")["data"].first || {}
+      end
+    },
+
+    get_related_balance_transactions: {
+      description: "Get related <span class='provider'>balance transactions" \
+        "</span> in <span class='provider'>Stripe</span>",
+
+      input_fields: lambda do
+        [
+          {
+            name: "payout",
+            label: "Payout ID",
+            optional: false
+          }
+        ]
+      end,
+
+      execute: lambda do |_connection, input|
+        get("/v1/balance/history?limit=100").
+          params(input)
+      end,
+
+      output_fields: lambda do |object_definitions|
+        [
+          { name: "data", type: "array",
+            properties: object_definitions["balance_transaction"] }
+        ]
+      end,
+
+      sample_output: lambda do |_connection|
+        get("/v1/balance/history?limit=1") || {}
+      end
+    },
+
+    get_refund: {
+      description: "Get <span class='provider'>refund</span> details in " \
+        "<span class='provider'>Stripe</span>",
+
+      input_fields: lambda do
+        [
+          {
+            name: "id",
+            label: "Refund ID",
+            optional: false
+          }
+        ]
+      end,
+
+      execute: lambda do |_connection, input|
+        get("/v1/refunds/#{input['id']}")
+      end,
+
+      output_fields: lambda do |object_definitions|
+        object_definitions["refund"]
+      end,
+
+      sample_output: lambda do |_connection|
+        get("/v1/refunds?limit=1")["data"].first || {}
+      end
+    }
+  },
+
+  triggers: {
+    new_customer: {
+      description: "New <span class='provider'>customer</span> in " \
+        "<span class='provider'>Stripe</span>",
+
+      type: "paging_desc",
+
+      input_fields: lambda do
+        [
+          {
+            name: "since",
+            label: "Created after",
+            type: "date_time",
+            control_type: "date_time",
+            hint: "Retrieve customers created after this date. Leave blank " \
+              "to retrieve all customers",
+            optional: true,
+            sticky: true
+          }
+        ]
+      end,
+
+      poll: lambda do |_connection, input, next_page|
+        starting_after = next_page
+
+        if starting_after.present?
+          response = get("/v1/customers?limit=100").
```

Comment:
This is not right. You will end up picking all records and ignore the since input.

Reply:
when poll is first started, wouldn't starting_after be nil and thus append the since input in the else block?

Reply:
In successive polls in the same interval, `starting_after` will be assigned some value which will drop the since parameter. In the case of multi-paged responses, this page through all records, regardless of the since parameter because as this code pages, it starts ignoring the since parameter on the 2nd page onwards.